src/shared/util.js

```diff
@@ -1654,6 +1632,623 @@ function loadJpegStream(id, imageUrl, objs) {
   img.src = imageUrl;
 }
 
+// Polyfill from https://github.com/Polymer/URL
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+(function checkURLConstructor(scope) {
+  /* jshint ignore:start */
+
+  // feature detect for URL constructor
+  var hasWorkingUrl = false;
+  if (!scope.forceJURL) {
```

Comment:
let's replace this with `typeof URL === 'function' && ('origin' in URL.prototype)`